{{cookiecutter.project_slug}}/clients/web/vue3/src/services/users/api.js

```diff
@@ -1,81 +1,47 @@
-import { ModelAPI, objectToCamelCase } from '@thinknimble/tn-models'
-
-import AxiosClient from '../AxiosClient'
-import { apiErrorHandler } from '../api'
-
-// NOTE: The AxiosClient is congfigured to include '/api' in the baseUrl
-const LOGIN_ENDPOINT = '/login/'
-const PASSWORD_RESET_EMAIL_ENDPOINT = '/password/reset/'
-const PASSWORD_RESET_ENDPOINT = '/password/reset/confirm/'
-const REGISTRATION_ENDPOINT = '/users/'
-const USERS_ENDPOINT = '/users/'
-
-export default class UserAPI extends ModelAPI {
-  /**
-   * ModelAPI contains methods for list and create (overridden here) and the FILTERS_MAP
-   * You may override any of these methods by statically defining them here
-   * e.g static FILTERS_MAP={...UserAPI.FILTERS_MAP, <FITERS>}
-   *      list({ filters = {}, pagination = {} }){
-   *
-   * }
-   */
-
-  get client() {
-    return AxiosClient
-  }
-
-  static ENDPOINT = USERS_ENDPOINT
-
-  login(d) {
-    const data = { email: d.email.toLowerCase(), password: d.password }
-    return this.client
-      .post(LOGIN_ENDPOINT, data)
-      .then((response) => this.cls.fromAPI(response.data))
-      .catch(
-        apiErrorHandler({
-          apiName: 'UserAPI.login',
-          enable400Alert: false,
-          enable500Alert: false
-        }),
-    )
-  }
-
-  registerUser(d) {
-    const data = {
-      firstName: d.firstName,
-      lastName: d.lastName,
-      email: d.email.toLowerCase(),
-      password: d.password,
-    }
-    return this.client
-      .post(REGISTRATION_ENDPOINT, this.cls.toAPI(data))
-      .then((response) => this.cls.fromAPI(response.data))
-      .catch(
-        apiErrorHandler({
-          apiName: 'UserAPI.registerUser',
-          enable400Alert: false,
-          enable500Alert: false,
-        }),
-      )
-  }
-
-  requestPasswordReset(email) {
-    const data = { email: email }
-    return this.client
-      .post(PASSWORD_RESET_EMAIL_ENDPOINT, data)
-      .catch(
-      apiErrorHandler({
-        apiName: 'UserAPI.requestPasswordReset',
-      }),
-    )
-  }
-
-  resetPassword({ uid, token, password }) {
-    const url = `${PASSWORD_RESET_ENDPOINT}${uid}/${token}/`
-    const data = { password }
-    return this.client
-      .post(url, data)
-      .then((response) => this.cls.fromAPI(response.data))
-      .catch(apiErrorHandler({ apiName: 'UserAPI.resetPassword' }))
-  }
-}
+import { createApi, createCustomServiceCall } from '@thinknimble/tn-models-fp'
+import { z } from 'zod'
+import axiosInstance from '../AxiosClient'
+import { userShape, forgotPasswordShape, userCreateShape, loginShape } from './models'
+
+const login = createCustomServiceCall(
+  {
+    inputShape: loginShape,
+    outputShape: userShape,
+  },
+  async ({ client, input, utils }) => {
+    const res = await client.post('api/login/', utils.toApi(input))
```

Comment:
I think we undid that one change, but all these URLs still start with `/api`. That might be why Cypress is failing

Reply:
yep we made the change but I  thought we said it doesn't matter in the end!